.github/actions/setup-python/action.yml

```diff
@@ -0,0 +1,60 @@
+name: 'python setup'
+description: 'Configures python, pip, and configures pip cache dir as output.'
+
+inputs:
+  cache-files-hash:
+    description: 'A single hash for a set of files. Used for caching.'
+    required: false
+    default: ${{ hashFiles('requirements-*.txt', '!requirements-pre-commit.txt') }}
+
+outputs:
+  python-version:
+    description: 'The version of python we are using'
+    value: ${{ steps.python-version.outputs.python-version }}
+  pip-cache-dir:
+    description: 'Path to pip cache'
+    value: ${{ steps.pip-info.outputs.pip-cache-dir }}
+  pip-version:
+    description: 'Pip version'
+    value: ${{ steps.pip-info.outputs.pip-version }}
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Set python version output
+      id: python-version
+      shell: bash
+      run: |
+        echo "::set-output name=python-version::$(cat .python-version)"
+
+    - name: Setup default environment variables
+      shell: bash
+      run: |
+        echo "PIP_DISABLE_PIP_VERSION_CHECK=on" >> $GITHUB_ENV
+
+    - uses: actions/setup-python@v2
+      env:
+        PIP_DISABLE_PIP_VERSION_CHECK: on
+      with:
+        python-version: ${{ steps.python-version.outputs.python-version }}
+
+    - name: Install pip
+      shell: bash
+      run: |
+        make upgrade-pip
+
+    - name: Get pip cache dir & version
+      id: pip-info
+      shell: bash
+      # pip versions before 20.1 do not have `pip cache`, `make upgrade-pip` ensures that version as a minimum
+      run: |
+        echo "::set-output name=pip-cache-dir::$(pip cache dir)"
+        echo "::set-output name=pip-version::$(pip -V | awk -F ' ' '{print $2}')"
+        echo "${{ inputs.cache-files-hash }}"
+
+    - name: pip cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.pip-info.outputs.pip-cache-dir }}
+        key: |
+          ${{ runner.os }}-py${{ steps.python-version.outputs.python-version }}-pip${{ steps.pip-info.outputs.pip-version }}-${{ inputs.pip-cache-version }}-${{ inputs.cache-files-hash }}
```

Comment:
Since we started pinning the version of `pip` we don't need to include it here in the key anymore.
It made a lot more sense in the dev env workflow than everywhere else.

I see `cache-files-hash` defined at the beginning of this file but I don't see `pip-cache-version`. I believe you told me once what would happen in such cases but I forget.

Reply:
Good catch, I forgot to include/pass `pip-cache-version` from setup-sentry. (It still works if you don't specify it under `inputs`).

I think we should still include pip version in the cache key as we would then wipe the cache when we upgrade pip.